libqtile/xcbq.py

```diff
@@ -316,20 +317,20 @@ def alloc_color(self, color):
         """
             Flexible color allocation.
         """
-        if color.startswith("#"):
-            if len(color) != 7:
+        try:
+            return self.conn.conn.core.AllocNamedColor(
+                self.cid, len(color), color
+            ).reply()
+        except xcffib.xproto.NameError:
+            if len(color) < 6:
                 raise ValueError("Invalid color: %s" % color)
```

Comment:
I do think the changes @kynikos has suggested would make this a little more readable, if you can make those the rest of it looks good to me.